### PR TITLE
feat: use ES modules

### LINKED ISSUE #1006: Incompatibility with Node.js v22.x (Assertion failed in esm)
Thank you for taking the time to look into this issue.

**Describe the bug**
When running shipjs prepare with Node.js v22.x, an assertion error occurs in the esm package, causing the process to fail. This issue does not occur when using Node.js v20.x

**To Reproduce**

Install Node.js v22.x (e.g., v22.11.0).
Run yarn install in a project that uses shipjs.
Execute yarn release:prepare.
Expected behavior
The shipjs prepare command should complete successfully, as it does in the Node.js v20.x environment, without any errors.

**Screenshots**
```
$ shipjs prepare

  #  /Users/******/.nvm/versions/node/v22.11.0/bin/node[69934]: void node::fs::InternalModuleStat(const FunctionCallbackInfo<v8::Value> &) at ../src/node_file.cc:1037
  #  Assertion failed: (args.Length()) >= (2)

----- JavaScript stack trace -----

1: /path/to/project/node_modules/esm/esm.js:1:34535
2: /path/to/project/node_modules/esm/esm.js:1:34176
3: /path/to/project/node_modules/esm/esm.js:1:34506
4: /path/to/project/node_modules/esm/esm.js:1:173374
5: /path/to/project/node_modules/esm/esm.js:1:173420
...
```
**Environment (please complete the following information):**
Node.js version: v22.11.0
Yarn version: v1.22.22
Ship.js version: v0.26.4

**Additional context**

This issue is likely caused by the dependency esm@3.2.25 used by shipjs, which might not fully support Node.js v22.x due to changes in its internal API.
Here is the result of yarn why esm:

```
=> Found "esm@3.2.25"
info Reasons this module exists
   - "shipjs" depends on it
   - Hoisted from "shipjs#esm"
```
This issue does not occur in Node.js v20.x (LTS: Iron). It would be greatly appreciated if compatibility with the latest Node.js v22.x (LTS: Jod) could be ensured.

### PR DESCRIPTION
**Pull request by JasonBerry**
* removed the reliance on the [esm](https://github.com/standard-things/esm/) library which has been dead for some time
* updated to Node 18, which is the LTS version [still being maintained](https://github.com/nodejs/Release) - Node 12 was EOL'd 2022-04-30
* tried to modify as little as possible to keep the PR small

Fixes #1006

BREAKING CHANGE: code is in in ES Modules now and earliest node support is 18

### CODE REVIEW

File: packages/shipjs/bin/shipjs [hunk 1]
```diff
@@ -1,10 +1,10 @@
 #!/usr/bin/env node
 
-require = require('esm')(module);
-(async function() {
+(async function () {
```
Comment (Haroenv) [inline packages/shipjs/bin/shipjs:3]: not sure about the history but this likely was an async iife to ensure all errors are caught. Not sure that's needed really tbh. Can be changed in a separate PR and release to be just a regular sync module maybe?

File: packages/shipjs/src/step/prepare/__tests__/createPullRequest.spec.js [hunk 2]
```diff
@@ -69,15 +69,8 @@ describe('createPullRequest', () => {
       .mockImplementation(({ version, type }) => `# v${version} (${type})`);
     const mockFormatPullRequestMessage = jest
       .fn()
-      .mockImplementation(
-        ({ formatPullRequestTitle, nextVersion, releaseType }) => {
-          return [
-            formatPullRequestTitle({
```
Comment (Haroenv) [inline packages/shipjs/src/step/prepare/__tests__/createPullRequest.spec.js:75]: why this change by the way? was the formatting not needed in the test?

**Review by Haroenv — APPROVED** [review]
checked that all files are correctly imported with seeing no matches for `from '\./.*[^js]';`, and runs correctly. thanks!

File: packages/shipjs/bin/shipjs (hunk 1 above)
Comment (JasonBerry) [reply packages/shipjs/bin/shipjs:3]: I'm easy with however you want to do it. I was just trying to keep changes to a minimum!

File: packages/shipjs/src/step/prepare/__tests__/createPullRequest.spec.js (hunk 2 above)
Comment (JasonBerry) [reply packages/shipjs/src/step/prepare/__tests__/createPullRequest.spec.js:75]: I don't know if/how this was ever working!

In `createPullRequest.js` `formatPullRequestTitle` is never passed to `formatPullRequestMessage` and so `formatPullRequestTitle` is `undefined` and not able to be called. The relevant bit is:
```js
    const title = formatPullRequestTitle({ version: nextVersion, releaseType });
    const message = formatPullRequestMessage({
      repo,
      repoURL,
      baseBranch,
      stagingBranch,
      destinationBranch: baseBranch,
      releaseType,
      diffURL,
      currentVersion,
      nextVersion,
      publishCommands,
      title,
    });
```

It doesn't make any sense that it was ever working.

File: packages/shipjs/src/step/prepare/__tests__/createPullRequest.spec.js [hunk 3]
```diff
@@ -69,15 +69,8 @@ describe('createPullRequest', () => {
       .mockImplementation(({ version, type }) => `# v${version} (${type})`);
```
Comment (JasonBerry) [inline packages/shipjs/src/step/prepare/__tests__/createPullRequest.spec.js:69]: This should actually be:
```js
.mockImplementation(({ version, releaseType }) => `# v${version} (${releaseType})`);
```
